File: javascript/rna-transcription/rna-transcription.js
var dnaTranscriber = function() {
	this.map = {
		'G': 'C',
		'C': 'G',
		'T': 'A',
		'A': 'U'
	}
	this.toRna = function(strand) {
		var rna = ""
		for(var i = 0; i < strand.length; i++) {
			if(!this.map[strand[i]] ) {
				throw Error('Invalid input');
			}
			rna += this.map[strand[i]];
		}
		return rna;
	}

}

module.exports = dnaTranscriber;

File: python/rna-transcription/rna_transcription.py
def to_rna(dna):
    xlate = {
            'G': 'C',
            'C': 'G',
            'T': 'A',
            'A': 'U'
            }
    rna  = ""
    for c in dna:
        if c in xlate.keys():
            rna += xlate[c]
        else:
            return("")
    return rna
